ggshield/cmd/iac/scan/ci.py

```diff
@@ -44,13 +48,30 @@ def scan_ci_cmd(
         return display_iac_scan_all_result(ctx, directory, result)
 
     config: Config = ctx.obj["config"]
-    commit_list, _ = collect_commit_range_from_ci_env(config.user_config.verbose)
-    reference, current_ref = commit_list[0], commit_list[-1]
 
-    # If we failed to fetch a current reference, we set it to HEAD
-    if len(commit_list) < 2 or not current_ref:
+    new_commits, _ = collect_commit_range_from_ci_env(config.user_config.verbose)
+    current_ref = new_commits[-1]
+    reference, _ = get_previous_commit_from_ci_env(config.user_config.verbose)
+
+    # If no sha provided, consider repo was empty before
+    if reference == EMPTY_SHA or reference == "null":
+        reference = None
+
+    # If we fail to fetch current state, we set it to HEAD
+    if current_ref == EMPTY_SHA or current_ref == "null":
         current_ref = "HEAD"
 
+    if config.user_config.verbose:
+        if len(new_commits) > 0:
+            click.echo("List of new commits: ")
```

Comment:
```suggestion
            click.echo("List of new commits:", err=True)
```

(or use `core.text_utils.display_info()`)